### ISSUE TITLE
Boot_serial: Stay in serial recovery until kicked from other app

### ISSUE THREAD

**Issue by de-nordic** [mcu-tools/mcuboot#1835]
**Enhancement**
MCUboot could be compiled in mode where it would stay in boot serial until it gets continuation trigger from other application either by UART or some specific pin.
This may be useful to be able to control boot process of MCU from other MCU (or core) that wants to be sure whether app that will be running on that core is proper.
For example when the board starts, the Main MCU starts MCUboot and goes int application. Then it queries other cores/MCUs connected to for image versions. If needed the main app uploads new versions or triggers image swaps, or if everything is ok it passes continue command that continues execution of selected boot application without resetting other core.

**In short**
MCUboot should be able to always stay in boot serial until it receives "continue" command.
